src/lib/errors.ts

```diff
@@ -10,7 +10,11 @@ export interface StatusCodeError extends Error {
 }
 
 export class StatusError extends Error {
```

Comment:
Can we change this class to be called `HttpError` because StatusError doesn't make sense. I saw that its only referenced in 6 places so should be easy to refactor.

Reply:
I think it makes sense as is tbh. `StatusError` is a node error that can also have a status code, we use it some places to provide more info about the error. 

`HttpError` is one class of status error, although I agree that adding the `retryAfter` moves it towards HTTP camp. 

Maybe I could create an `HttpError` that extends `StatusError` with the `retryAfter` and is used by the current state report, but I don't see it as a urgent improvement.

Reply:
if StatusError is used in places that are not related to HTTP then that makes sense. Extending it seems nice but we can leave it for now I suppose since it's only 1 spot.